app/main.py

```diff
@@ -1,3 +1,77 @@
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: (float, int)) -> None:
+        self.km = km
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __add__(self, other: (object, float, int)) -> object:
+        if isinstance(other, (float, int)):
+            create_object_from_int = Distance(other)
+            return Distance(self.km + create_object_from_int.km)
+
+        return Distance(self.km + other.km)
+
+    def __iadd__(self, other: (object, float, int)) -> object:
+        if isinstance(other, (float, int)):
+            create_object_from_int = Distance(other)
+            self.km = self.__add__(create_object_from_int).km
+
+            return self
+
+        self.km = self.__add__(other)
```

Comment:
In the `__iadd__` method, `self.km` is being assigned the result of `self.__add__(other)`, which is a `Distance` object. This is incorrect because `self.km` should be a numeric value (float or int). You should extract the `km` value from the result of `__add__`.